regression/tests/regression_test_device.cpp

```diff
@@ -0,0 +1,225 @@
+    ASSERT_EQ(domain.getTickResolution().getDenominator(), 1000000);
+    ASSERT_EQ(domain.getOrigin(), "1970-01-01T00:00:00Z");
+
+    if (protocol == "opcua")
+    {
+        ASSERT_EQ(domain.getUnit(), Unit("", -1, "", ""));
+    }
+    else if (protocol == "nd")
+    {
+        ASSERT_EQ(domain.getUnit(), Unit("s", -1, "second", "time"));
+    }
+}
+
+TEST_F(RegressionTestDevice, getInputsOutputsFolder)
+{
+    FolderPtr folder;
+    ASSERT_NO_THROW(folder = device.getInputsOutputsFolder());
+    if (protocol == "opcua" || protocol == "nd")
+    {
+        ASSERT_EQ(folder.getItems().getCount(), 2);
+    }
+    else if (protocol == "ns" || protocol == "lt")
+    {
+        ASSERT_EQ(folder.getItems().getCount(), 0);
+    }
+}
+
+TEST_F(RegressionTestDevice, getCustomComponents)
+{
+    ListPtr<IComponent> components;
+    ASSERT_NO_THROW(components = device.getCustomComponents());
+    if (protocol == "opcua" || protocol == "nd")
+    {
+        ASSERT_EQ(components.getCount(), 1);
+    }
+    else if (protocol == "ns" || protocol == "lt")
+    {
+        ASSERT_EQ(components.getCount(), 0);
+    }
+}
+
+TEST_F(RegressionTestDevice, getSignals)
+{
+    ListPtr<ISignal> signals;
+    ASSERT_NO_THROW(signals = device.getSignals());
+    if (protocol == "opcua" || protocol == "nd")
+    {
+        ASSERT_EQ(signals.getCount(), 0);
+    }
+    else if (protocol == "ns")
+    {
+        ASSERT_EQ(signals.getCount(), 4);
+    }
+    else if (protocol == "lt")
+    {
+        ASSERT_EQ(signals.getCount(), 6);
+    }
+}
+
+TEST_F(RegressionTestDevice, getSignalsRecursive)
+{
+    ListPtr<ISignal> signals;
+    ASSERT_NO_THROW(signals = device.getSignalsRecursive());
+    if (protocol == "opcua" || protocol == "nd" || protocol == "ns")
+    {
+        ASSERT_EQ(signals.getCount(), 4);
+    }
+    else if (protocol == "lt")
+    {
+        ASSERT_EQ(signals.getCount(), 6);
+    }
+}
+
+TEST_F(RegressionTestDevice, getChannels)
+{
+    ListPtr<IChannel> channels;
+    ASSERT_NO_THROW(channels = device.getChannels());
+    if (protocol == "opcua" || protocol == "nd")
+    {
+        ASSERT_EQ(channels.getCount(), 2);
+    }
+    else if (protocol == "ns" || protocol == "lt")
+    {
+        ASSERT_EQ(channels.getCount(), 0);
+    }
+}
+
+TEST_F(RegressionTestDevice, getChannelsRecursive)
+{
+    ListPtr<IChannel> channels;
+    ASSERT_NO_THROW(channels = device.getChannelsRecursive());
+    if (protocol == "opcua" || protocol == "nd")
+    {
+        ASSERT_EQ(channels.getCount(), 2);
+    }
+    else if (protocol == "ns" || protocol == "lt")
+    {
+        ASSERT_EQ(channels.getCount(), 0);
+    }
+}
+
+TEST_F(RegressionTestDevice, getDevices)
+{
+    ListPtr<IDevice> devices;
+    ASSERT_NO_THROW(devices = device.getDevices());
+    ASSERT_EQ(devices.getCount(), 0);
+}
+
+TEST_F(RegressionTestDevice, getAvailableDevices)
+{
+    ListPtr<IDeviceInfo> infos;
+    ASSERT_NO_THROW(infos = device.getAvailableDevices());
```

Comment:
we can check discovering by enabling mdns in regression simultor generator. the example of enabling is in [quick_start_simulator](https://github.com/openDAQ/openDAQ/blob/main/examples/cpp/quick_start/quick_start_simulator.cpp)

Reply:
Thank you for your comment, but I don't think this is possible in the version we are currently targeting with the regression tests (the version of the SDK that the regression simulator runs on), which is [tag v3.0.0](https://github.com/openDAQ/openDAQ/tree/v3.0.0). I think the best course of action is to test discovery on a later version. There are already plans to target different versions of the SDK with regression tests anyways, and this is on the top of the TODO list for further expansion of the regression test framework for sure.